### PR TITLE
Use Bazel to build Docker images

### PR DESCRIPTION
**Pull request by kyessenov**
With this change we get deterministic and consistent docker builds for Manager and Proxy.
Try this:
    
    bazel run //docker:runtime

This change is [Reviewable](https://reviewable.io/reviews/istio/manager/51)

### CODE REVIEW

File: proxy/envoy/agent.go [hunk 1]
```diff
@@ -44,39 +46,21 @@ type instance struct {
...
+		binary:     binary,
+		configRoot: configRoot,
+		cmdMap:     make(map[*exec.Cmd]instance),
+	}
 }
 
-func configFile(epoch int) string {
-	return fmt.Sprintf(EnvoyConfigPath+EnvoyFileTemplate, epoch)
+func configFile(config string, epoch int) string {
+	return fmt.Sprintf(config+EnvoyFileTemplate, epoch)
```
Comment (douglas-reid) [inline proxy/envoy/agent.go]: any reason to not just have the EnvoyFileTemplate include a %s at the beginning and pass config as an arg here?

File: proxy/envoy/config.go [hunk 2]
```diff
@@ -106,7 +118,9 @@ func Generate(instances []*model.ServiceInstance, services []*model.Service, mes
 
 // buildListeners uses iptables port redirect to route traffic either into the pod or outside the pod
 // to service clusters based on the traffic metadata.
-func buildListeners(instances []*model.ServiceInstance, services []*model.Service) ([]Listener, []Cluster) {
+func buildListeners(instances []*model.ServiceInstance,
+	services []*model.Service,
```
Comment (douglas-reid) [inline proxy/envoy/config.go]: curious: why spread this out across lines?

File: proxy/envoy/agent.go (hunk 1 above)
Comment (kyessenov) [inline proxy/envoy/agent.go]: Makes sense. The config is always meant to be a path prefix, so maybe it's easier to enforce that with concatenation?

File: proxy/envoy/config.go (hunk 2 above)
Comment (kyessenov) [inline proxy/envoy/config.go]: `lll` barked at me... I'll fix this.

**Pull request merged**